libs/application/templates/parental-leave/src/constants.ts

```diff
@@ -98,6 +104,29 @@ export enum AnswerValidationConstants {
   // the repeater sends all the periods saved in 'periods'
   // to this validator, which will validate the latest one
   VALIDATE_LATEST_PERIOD = 'periods',
+  RESIDENCE_GRANT = 'residenceGrant',
 }
 
 export const DATE_FORMAT = 'yyyy-MM-dd'
+
+export type Events =
+  | { type: DefaultEvents.APPROVE }
+  | { type: DefaultEvents.ASSIGN }
+  | { type: DefaultEvents.REJECT }
+  | { type: DefaultEvents.SUBMIT }
+  | { type: DefaultEvents.ABORT }
+  | { type: DefaultEvents.EDIT }
+  | { type: 'MODIFY' } // Ex: The user might modify their 'edits'.
+  | { type: 'CLOSED' } // Ex: Close application
+  /**
+   * States for routing Dvalarstyrkur
+   *  Takes previous state and add
+   *  a postfix of REJECT if rejected button is pushed
+   */
+  | { type: 'APPROVED' }
+  | { type: 'RESIDENCEGRANTAPPLICATION' } // Ex: when the baby is born a parent can apply for resident grant
+  | { type: 'ADDITIONALDOCUMENTSREQUIRED' } // Ex: VMST ask for more documents
+  | { type: 'APPROVEDREJECT' }
+  | { type: 'VINNUMALASTOFNUNAPPROVALREJECT' }
+  | { type: 'RESIDENCEGRANTAPPLICATIONREJECT' }
+  | { type: 'VINNUMALASTOFNUNAPPROVEEDITSREJECT' }
```

Comment:
Is it really necessary to add more types ( APPROVED and REJECT ) here insteads of use DefaultEvents types?

Reply:
Could we use Approve/Reject with condition like in Draft state?

Reply:
This has been changed